sospice/util/sigma_clipping.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from scipy.ndimage import generic_filter
+from numpy import ma
+
+
+def _stddev_loc(image, size):
+    if type(size) is int:
+        size = (size,) * image.ndim
+    mean2 = _blur(image, size) ** 2
+    vari = _blur(image ** 2, size)
+    vari -= mean2
+    vari[vari <= 0] = 1e-20
+    return np.sqrt(vari)
+
+
+def _blur(image, size, output=None):
+
+    if type(size) is int:
+        size = (size,)*image.ndim
+
+    if output is None:
+        output = np.empty_like(image)
+
+    if image.ndim == 2:
+        generic_filter(image, np.nanmean, mode='reflect', output=output, size=size)
+    elif image.ndim == 3:
+        for i in range(image.shape[0]):
+            if size[1] > 1 or size[2] > 1:
+                generic_filter(image[i], np.nanmean, mode='reflect', output=output[i], size=size[2:0:-1])
+            else:
+                output[i] = np.copy(image[i])
+        for i in range(image.shape[2]):
+            if size[0] > 1:
+                dum = np.empty_like(output[:, :, i])
+                generic_filter(np.copy(output[:, :, i]), np.nanmean, mode='reflect', output=dum, size=(1, size[0]))
+                output[:, :, i] = dum
+    else:
+        raise ValueError('Unsupported number of dimensions')
+
+    return output
+
+
+def _get_center_function(image, center):
+    if center not in ['mean', 'median']:
+        raise ValueError('Center must be mean of median')
+    if np.sum(image.mask) > 0:
```

Comment:
If I understand correctly, `generic_filter()` completely ignores masks in masked arrays. Then, what is relevant: is it the number of masked pixels (`np.sum(image.mask)`), the fact that there is any masked pixel (`image.mask.any()`), or the fact that there is any NaN (`image.isnan().any()`)?